lib/utils/report.mjs

```diff
@@ -213,27 +218,94 @@ function isDisabled(ruleName, startLine, disabledRanges) {
 	return false;
 }
 
-/** @param {Problem & { line: number }} problem */
+/**
+ * @param {Problem & { line: number }} problem
+ * @returns {boolean}
+ */
 function isFixApplied({ fix, line, result: { stylelint }, ruleName }) {
+	if (!fix) return false;
+
 	const { disabledRanges, config = {}, fixersData } = stylelint;
 
-	if (!isFn(fix)) {
-		return false;
-	}
+	if (!config.fix) return false;
+
+	if (isFixDisabled(line, ruleName, config, disabledRanges)) return false;
 
-	const shouldFix = Boolean(config.fix && !config.rules?.[ruleName][1]?.disableFix);
-	const mayFix =
-		shouldFix && (config.ignoreDisables || !isDisabled(ruleName, line, disabledRanges));
+	const apply = isFixObject(fix) ? fix.apply : fix;
 
-	if (!mayFix) return false;
+	if (!isFn(apply)) return false;
 
-	fix();
+	apply();
 
 	incrementFixCounter({ fixersData, ruleName });
 
 	return true;
 }
 
+/**
+ * @param {Problem & { line: number }} problem
+ * @returns {{range: [number, number], text: string} | undefined}
+ */
+function computeEditInfo({ fix, line, result: { stylelint }, ruleName }) {
+	if (!fix) return;
+
+	const { disabledRanges, config = {}, fixedRanges } = stylelint;
+
+	if (!config.computeReplacementText) return;
+
+	if (isFixDisabled(line, ruleName, config, disabledRanges)) return;
```

Comment:
nit
opinion

I think that either [`disableFix` documentation](https://stylelint.io/user-guide/configure/#disablefix) should be updated or that `disableFix` should be ignored by `computeEditInfo` function. I am not sure what's the best course of action though.

Reply:
Any user facing documentation still needs to happen.
I suggest we document that only those rules that are enabled both in general and for fixes will have computed edit info.

I want to avoid having a third mechanism to enable warnings while disabling some rules for edit info.